packages/utils/src/conversions.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * @module Conversions
+ * This module provides a collection of utility functions for converting
+ * between different numerical formats, particularly focusing on
+ * conversions involving bigints, hexadecimals and buffers.
+ * The module is structured with clear function naming to indicate
+ * the conversion direction (e.g., `bigIntToHexadecimal` for BigInt
+ * to hexadecimal, `bufferToBigint` for buffer to bigint) and employs
+ * type checks to ensure the correct handling of various input types.
+ * It also includes variations for both big-endian (`be`) and little-endian
+ * (`le`) conversions. It is important to note that when there is no prefix,
+ * the order of bytes is always big-endian.
+ */
+
+import { BigNumberish } from "./types"
+import { isHexadecimal, isStringifiedBigint } from "./number-checks"
+
+/**
+ * Converts a bigint to a hexadecimal string ensuring even length.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function beBigIntToHexadecimal(n: bigint): string {
+    let hex = n.toString(16)
+
+    // Ensure even length.
+    if (hex.length % 2 !== 0) {
```

Comment:
I don't know all the use cases, but I wonder if what they really need isn't padding to an even number, but padding out to a specific size (e.g. 32 bytes) which should be passed as an argument.  This doesn't need to be resolved in this PR, it's just something which occurs to me as an easy misuse.

Reply:
From what I know `Buffer` needs it. `Buffer.from("02", "hex")` is not the same as `Buffer.from("2", "hex")`.

It shouldn't make any difference in the other use-cases.

Feel free to open another issue for that feature if you want 👍🏽

Reply:
BTW there's an issue for replacing `Buffer` with `Uint8Array`, to make these libraries fully-compatible with browsers -> https://github.com/privacy-scaling-explorations/zk-kit/issues/144.

Your input would be really appreciated. I'll work on that asap.